cr-interactors/src/main/java/com/thefuntasty/mvvm/crinteractors/CoroutineScopeOwner.kt

```diff
@@ -73,22 +73,21 @@ interface CoroutineScopeOwner {
         onComplete: () -> Unit = {}
     ) {
         job?.cancel()
-        job = coroutineScope.launch(Dispatchers.Main) {
-            try {
-                build()
-                    .flowOn(getWorkerDispatcher())
-                    .onEach { onNext(it) }
-                    .onCompletion { error ->
-                        if (this@launch.isActive) {
-                            error?.also { onError?.invoke(it) ?: throw error } ?: onComplete()
-                        }
+        job = build()
+            .flowOn(getWorkerDispatcher())
+            .onEach { onNext(it) }
```

Comment:
Isn`t collect() method better practice to pass items to onNext() method, and also wouldn't this mean that onNext() is called on IO dispatcher?

Reply:
flowOn has impact on stream operations before this operator. Not after.

Reply:
to use collect() you have would have to wrap this code to new coroutine block. With onEach() method we can write this in stream-like syntax. Which is desired, since this is a stream.

Reply:
It is also related to the fact `collect()` is not used anymore. `launchIn()` is used instead.